### ISSUE TITLE
Crash - XInitThreads has not been called

### ISSUE THREAD

**Issue by Lexus89** [rofl0r/MacGeiger#1]
When i run the tool inside an Ubuntu VM (VMware Workstation) it crashes and displays the following error:

```
[xcb] Unknown sequence number while processing queue
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
macgeiger: ../../src/xcb_io.c:274: poll_for_event: Assertion `!xcb_xlib_threads_sequence_lost' failed.
Aborted
```

It runs for a few seconds and then stops.

UPDATE:

When running a few times more i also get this:

```
XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":0"
      after 116 requests (116 known processed) with 2 events remaining.
```

**Comment by rofl0r** [comment]
yeah, i have the same issue when i use the SDL backend. it looks like its some bug in SDL 1.x which is triggered when several threads are active.

IIRC it first happened after this change 45541024ec3bbfcede8693886881945ee417cc2a

the only known workaround at this time is to either revert to before that commit or use the NCURSES terminal backend.

**Comment by Lexus89** [comment]
Thanks, with NCURSES all is working fine. Great tool!

**Comment by rofl0r** [comment]
meanwhile i developed a patch to fix this issue, see https://github.com/sabotage-linux/sabotage/commit/261c52650c033e82ea088a21df67ba0716ed3b62

since SDL 1.2 development has officially been halted, there's little chance this will ever make it upstream though.

related:
https://bugzilla.libsdl.org/show_bug.cgi?id=2843
https://bugzilla.redhat.com/show_bug.cgi?id=1181085

i'll leave this open as the issue ain't really fixed until this goes into SDL itself.
